engine/src/cmd/dock_utils.cpp

```diff
@@ -0,0 +1,200 @@
+        if (range < 0) {
+            return 0;
+        } else {
+            return -1;
+        }
+    } 
+
+
+    if(configuration()->dock.simple_dock) {
+        range = DistanceTwoTargets(dock, ship);
+
+        if (range < 5000) {
+            return 0;
+        } else {
+            return -1;
+        }
+    }
+
+    if (ignore_occupancy) {
+        for (unsigned int i = 0; i < dock->pImage->dockingports.size(); ++i) {
+            if (!dock->pImage->dockingports[i].IsOccupied()) {
+                return i;
+            }
+        }
+    }
+
+    //don't need to check relation: already cleared.
+
+    // If your unit has docking ports then we check if any of our docking
+    // ports overlap with any of the station's docking ports.
+    // Otherwise we simply check if our unit overlaps with any of the
+    // station's docking ports.
+    for (unsigned int i = 0; i < dock->pImage->dockingports.size(); ++i) {
+        if (!dock->pImage->dockingports.empty()) {
+            for (unsigned int j = 0; j < dock->pImage->dockingports.size(); ++j) {
+                if (insideDock(dock->pImage->dockingports[i],
+                        InvTransform(dock->GetTransformation(),
+                                Transform(dock->GetTransformation(),
+                                        dock->pImage->dockingports[j].GetPosition().Cast())),
+                        dock->pImage->dockingports[j].GetRadius())) {
+                    // We cannot dock if we are already docked
+                    if (((dock->docked & (dock->DOCKED_INSIDE | dock->DOCKED)) == 0) && (!(dock->docked & dock->DOCKED_INSIDE))) {
+                        return i;
+                    }
+                }
+            }
+        } else if (insideDock(dock->pImage->dockingports[i],
+                InvTransform(dock->GetTransformation(), dock->Position()),
+                dock->rSize())) {
+            return i;
+        }
+    }
+    
+    return -1;
+}
+
+std::string GetDockingText(Unit *unit, Unit *target, double range) {
+    //If I can dock, don't bother displaying distance
+    if (CanDock(target, unit, false) != -1) {
+        return std::string("Docking: Ready");
+    }
+    
+    
+    // Planets/non-planets calculate differently
+    if (target->isUnit() == Vega_UnitType::planet) {
+        // TODO: move from here. We shouldn't have kill and land logic here.
+        if(range < 0) {
+            unit->hull.Destroy();
+        }
+
+        range -= target->rSize() * configuration()->dock.dock_planet_radius_percent;
+        if (range < 0) {
+            // Should be caught by dock check above
+            abort();
+        } else if (range < 100000) {
+            return std::string("Docking: ") + string(PrettyDistanceString(range));
+        }
+    } else {
+        if(configuration()->dock.simple_dock && target->pImage->dockingports.size() != 0 && 
+            range < 100000) {
+            if (range > 5000) {
```

Comment:
These numbers should be made constants, so that they are defined only once in a single place, this will avoid discrepancies, if only one instance is changed in the future. That would also make them easier to make them configurable, if we want that.

Reply:
I agree. But I was working under a time pressure to get a fix in 24 hours. I'll make the changes.

Reply:
Fixed.